nautobot/utilities/drf_views.py

```diff
@@ -231,48 +245,31 @@ def form_valid(self, request, obj, form, **kwargs):
                 msg = "Object import failed due to object-level permissions violation"
                 self.logger.debug(msg)
                 form.add_error(None, msg)
-        return self.form_invalid(request, obj, form)
 
-    def form_invalid(self, request, obj, form, **kwargs):
-        context={}
+        data = {}
+        if self.action == "bulk_edit" or self.action == "bulk_delete":
+            pk_list = kwargs.pop("pk_list")
+            table = self.table(self.queryset.filter(pk__in=pk_list), orderable=False)
+            if not table.rows:
+                messages.warning(
+                    request,
+                    f"No {self.queryset.model._meta.verbose_name_plural} were selected for deletion.",
+                )
+                return redirect(self.get_return_url(request))
+
+            data = {
+                "table": table,
+            }
+        data.update({"form": form})
+        return Response(data)
+
+    def form_invalid(self, form):
+        context = {}
+        context.update({"form": form})
         self.logger.debug("Form Validation Failed")
-        if self.action == "perform_destroy":
-            context.update(
-                {
-                    "obj": obj,
-                    "form": form,
-                    "obj_type": self.queryset.model._meta.verbose_name,
-                    "return_url": self.get_return_url(request, obj),
-                    "template": self.get_template_name("delete"),
-                }
-            )
-            return Response(context)
-        elif self.action == "perform_create_or_update":
-            context.update(
-                {
-                    "obj": obj,
-                    "form": form,
-                    "obj_type": self.queryset.model._meta.verbose_name,
-                    "return_url": self.get_return_url(request, obj),
-                    "editing": obj.present_in_database,
-                    "template": self.get_template_name("edit"),
-                }
-            )
-            return Response(context)
-        elif self.action == "perform_bulk_create":
-            context.update(
-                {
-                    "form": form,
-                    "fields": self.import_form().fields,
-                    "obj_type": self.import_form._meta.model._meta.verbose_name,
-                    "return_url": self.get_return_url(request),
-                    "active_tab": "csv-data",
-                    "template": self.get_template_name("bulk_import"),
-                }
-            )
-            return Response(context)
-        else:
-            pass
+        if self.action == "bulk_edit" or self.action == "bulk_destroy":
+            return context
```

Comment:
Why just returning a raw `context` here vs. wrapped in a `Response` in the case of the bulk update/delete actions?